admin/class-social-facebook.php

```diff
@@ -0,0 +1,349 @@
+						'<a href="' . esc_url( $this->options['fb_admins'][ $user_id ]['link'] ) . '">' . esc_html( $this->options['fb_admins'][ $user_id ]['name'] ) . '</a>'
+					),
+					'error'
+				);
+			}
+
+			unset( $user_id );
+		}
+
+		// Clean up the referrer url for later use
+		if ( filter_input( INPUT_SERVER, 'REQUEST_URI' ) ) {
+			$this->cleanup_referrer_url( 'key', 'userid', 'userrealname', 'link' );
+		}
+	}
+
+	/**
+	 * Clean up the request_uri. The given params are the params that will be removed from the URL
+	 */
+	private function cleanup_referrer_url() {
+		$_SERVER['REQUEST_URI'] = remove_query_arg(
+			func_get_args(),
+			filter_input(
+				INPUT_SERVER, 'REQUEST_URI', FILTER_CALLBACK, array( 'options' => 'sanitize_text_field' )
+			)
+		);
+	}
+
+	/**
+	 * When something is going well, show a success notice
+	 *
+	 * @param string $notice_text
+	 */
+	private function success_notice( $notice_text ) {
+		add_settings_error( 'yoast_wpseo_social_options', 'success', $notice_text, 'updated' );
+	}
+
+	/**
+	 * Verify the nonce from the URL with the saved nonce
+	 *
+	 * @param string $nonce_name
+	 */
+	private function verify_nonce( $nonce_name ) {
+		if ( wp_verify_nonce( filter_input( INPUT_GET, 'nonce' ), $nonce_name ) != 1 ) {
+			die( "I don't think that's really nice of you!." );
+		}
+	}
+
+	/**
+	 * Saving the options
+	 */
+	private function save_options() {
+		update_option( 'wpseo_social', $this->options );
+	}
+
+}
+
+/**
+ * This will display the HTML for the facebook insights part
+ */
+class Yoast_Social_Facebook_Form {
+
+	/**
+	 * @var	array	- The options for social
+	 */
+	private $options;
+
+	/**
+	 * @var array	- The repository for the buttons that will be shown
+	 */
+	private $buttons = array();
+
+	/**
+	 * @var string	- The URL to link to
+	 */
+	private $admin_url = 'admin.php?page=wpseo_social';
+
+	/**
+	 * Setting the options and call the methods to display everything
+	 *
+	 */
+	public function __construct() {
+		$this->options = get_option( 'wpseo_social' );
+	}
+
+	/**
+	 * Returns the output-property
+	 */
+	public function show_form() {
+		$this
+			->form_head()
+			->user_admin()
+			->show_buttons()
+			->app_admin();
+	}
+
+	/**
+	 * SHow the top of the social insights part of the page
+	 *
+	 * @return $this
+	 */
+	private function form_head() {
+		echo '<p><strong>' . esc_html__( 'Facebook Insights and Admins', 'wordpress-seo' ) . '</strong><br />';
+		echo sprintf(
+			esc_html__(
+				'To be able to access your %sFacebook Insights%s for your site, you need to specify a Facebook Admin. This can be a user, but if you have an app for your site, you could use that. For most people a user will be "good enough" though.', 'wordpress-seo'
+			),
+			'<a href="https://www.facebook.com/insights">',
+			'</a>'
+		);
+		echo'</p>';
+
+		return $this;
+	}
+
+	/**
+	 * Display the button to add an admin or add another admin from Facebook
+	 *
+	 * @return $this
+	 */
+	private function user_admin() {
```

Comment:
I think this function name isn't really descriptive. It shows a button, the function name should reflect that.